pdc/apps/auth/middleware.py

```diff
@@ -48,9 +49,12 @@ def process_request(self, request):
                 return
         # We are seeing this user for the first time in this session, attempt
         # to authenticate the user.
-        user = auth.authenticate(remote_user=username)
+        user = auth.authenticate(remote_user=username, request=request)
         if user:
             # User is valid.  Set request.user and persist user in the session
             # by logging the user in.
             request.user = user
-            auth.login(request, user)
+            request.session['auth_backend'] = user.backend
+            backend = load_backend(user.backend)
+            if not hasattr(backend, 'save_login') or not backend.save_login:
```

Comment:
I probably don't understand that. `KerberosUserBackend` does not have the `save_login` attribute, so condition is true. `AuthMellonUserBackend` has `save_login = False`, so with the negation the condition is also true. Is this intended?

Essentially, the condition is equivalent to `not getattr(backend, 'save_login', False)`, while I think it would make more sense as `getattr(backend, 'save_login', True)`. What am I missing?